lib/extensions/ar_set_application_name.rb

```diff
@@ -0,0 +1,14 @@
+module ArSetApplicationName
+  # Sets the current and future db connections' application
+  # name if the name is not known until after some startup
+  # and querying some tables.
+  #
+  # If we can determine the name before establish_connection
+  # time, it's probably better to module prepend that method
+  # and run "SET application_name TO 'my_app'"
+  def self.name=(name)
+    # TODO: this is postgresql specific
+    ENV['PGAPPNAME'] = name
+    ActiveRecord::Base.connection_pool.disconnect!
```

Comment:
Note, postgresql will look for an application name in this ENV variable, so just setting it before establishing a connection is enough to get this set.